spec/services/encrypted_key_maker_spec.rb

```diff
@@ -91,6 +91,18 @@
         allow(FeatureManagement).to receive(:use_kms?).and_return(true)
         expect(subject.unlock(user_access_key, encryption_key)).to eq hash_E
       end
+
+      it 'removes KMS ciphertext prefix before deciphering' do
+        tagged_ciphertext = EncryptedKeyMaker::KEY_TYPE[:KMS] + ciphered_key
+        expect(user_access_key).to receive(:store_encrypted_key).
+          with(tagged_ciphertext).and_call_original
+        expect(user_access_key).to receive(:unlock).with(random_R)
+
+        subject.make(user_access_key)
+        encryption_key = user_access_key.encryption_key
+
+        subject.unlock(user_access_key, encryption_key)
+      end
```

Comment:
you're right @monfresh -- I was stubbing the wrong thing. I needed to re-stub the aws client to expect a ciphertext with no prefix. Otherwise aws stubbed client just keeps returning a pre-defined good value, regardless of what it is passed.